vkcs/resource_vkcs_publicdns_zone.go

```diff
@@ -0,0 +1,227 @@
+package vkcs
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	zoneDelay         = 10 * time.Second
+	zoneMinTimeout    = 3 * time.Second
+	zoneCreateTimeout = 10 * time.Minute
+	zoneDeleteTimeout = 10 * time.Minute
+)
+
+const (
+	zoneStatusActive  = "active"
+	zoneStatusDeleted = "deleted"
+)
+
+func resourcePublicDNSZone() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourcePublicDNSZoneCreate,
+		ReadContext:   resourcePublicDNSZoneRead,
+		UpdateContext: resourcePublicDNSZoneUpdate,
+		DeleteContext: resourcePublicDNSZoneDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(zoneCreateTimeout),
+			Delete: schema.DefaultTimeout(zoneDeleteTimeout),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The region in which to obtain the V2 Public DNS client. If omitted, the `region` argument of the provider is used. Changing this creates a new zone.",
+			},
+
+			"primary_dns": {
+				Type:        schema.TypeString,
+				Optional:    true,
```

Comment:
really optional?

Reply:
Yes. Default value for it is "ns1.mcs.mail.ru".